app/packages/core/src/components/Schema/SchemaSettings.tsx

```diff
@@ -73,11 +73,28 @@ const SchemaSettings = () => {
     return null;
   }
 
+  const closeModal = (params: { resetStage: boolean }) => {
+    const { resetStage } = params;
+    setSettingsModal({ ...settingModal, open: false });
+    setSearchTerm("");
+    setSelectedPaths(originalSelectedPaths);
+    if (resetStage) {
+      setSelectedFieldsStage(null);
+    }
+  };
+
   return (
     <Fragment>
       <ModalWrapper
-        ref={schemaModalRef}
-        onClick={(event) => event.target === schemaModalRef.current}
+        ref={schemaModalWrapperRef}
+        onClick={(event) => {
+          const clickedOutsideModal =
+            event.target === schemaModalWrapperRef.current;
+          if (clickedOutsideModal) {
+            closeModal({ resetStage: false });
+          }
+          return clickedOutsideModal;
+        }}
```

Comment:
Sweet! You might want to use a pre-existing hook we have for this called `useOutsideClick(ref, handler)`